integrated_channels/integrated_channel/transmitters/course_metadata.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""
+Generic course metadata transmitter for integrated channels.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import json
+import logging
+
+from integrated_channels.integrated_channel.client import IntegratedChannelApiClient
+from integrated_channels.integrated_channel.transmitters import Transmitter
+from requests import RequestException
+
+from django.apps import apps
+from django.core.exceptions import ObjectDoesNotExist
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CourseTransmitter(Transmitter):
+    """
+    A generic course metadata transmitter.
+
+    It may be subclassed by specific integrated channel course metadata transmitters for
+    each integrated channel's particular course metadata transmission requirements and expectations.
+    """
+
+    def __init__(self, enterprise_configuration, client=IntegratedChannelApiClient):
+        """
+        By default, use the interface integrated channel API client which raises an error if used.
```

Comment:
@UmanShahzad Thanks for adding this highly confusing docstring ;) :P 

I stared at it for a couple minutes and I think I get it now, but perhaps there is a way to make it a bit more accessible? For example:

* Do something with "interface", e.g., replace it with "default" or "abstract" or something?
* At first glance it's not clear why the integrated channel API client would raise an error "if used".

Reply:
Yeah, I'll use 'abstract' as that's probably more in line with how we think about it in Python. Will handle the last point with some rewording too.